substrate/frame/scheduler/src/lib.rs

```diff
@@ -1192,6 +1322,52 @@ impl<T: Config> Pallet<T> {
 		let _ = weight.try_consume(call_weight);
 		Ok(result)
 	}
+
+	/// Check if a task has a retry configutation in place and, if so, try to reschedule it.
+	///
+	/// If the task was successfully rescheduled for later, this function will return `None`.
+	/// Otherwise, if the task was not rescheduled, either because there was no retry configuration
+	/// in place, there were no more retry attempts left, or the agenda was full, this function
+	/// will return the task so it can be handled somewhere else.
+	fn check_retry(
+		now: BlockNumberFor<T>,
+		when: BlockNumberFor<T>,
+		agenda_index: u32,
+		task: ScheduledOf<T>,
+	) -> Option<ScheduledOf<T>> {
```

Comment:
It does not make a big difference, but semantically it seems that the return type of this function is `Result<(), ScheduledOf<T>>`.

Reply:
Yes. Initially I thought that it's not really an error if there's no space so I was hesitant about the `Result`, but I changed my mind reading the code again :grin: .